x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/pipeline_select.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiSelectable, useIsWithinMaxBreakpoint } from '@elastic/eui';
+
+import { MLInferenceLogic, MLInferencePipelineOption } from './ml_inference_logic';
+import { PipelineSelectOption, PipelineSelectOptionProps } from './pipeline_select_option';
+
+export const PipelineSelect: React.FC = () => {
+  const {
+    addInferencePipelineModal: { configuration },
+    existingInferencePipelines,
+  } = useValues(MLInferenceLogic);
+  const { selectExistingPipeline } = useActions(MLInferenceLogic);
+
+  const { existingPipeline, pipelineName } = configuration;
+
+  const getPipelineOptions = (
+    pipelineOptions: MLInferencePipelineOption[]
+  ): PipelineSelectOptionProps[] => {
+    return pipelineOptions.map((pipelineOption) => ({
+      checked: existingPipeline && pipelineOption.pipelineName === pipelineName ? 'on' : undefined,
```

Comment:
isn't this `existingPipeline` check redundant? I thought this component is only rendered if `existingPipeline === true`

Reply:
Yes, it is, another one of my paranoid checks